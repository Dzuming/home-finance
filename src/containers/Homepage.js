import React, { Component } from 'react';
import { Card, CardContent, Grid, Typography, withStyles } from 'material-ui';
import Budget from './Budget';
import Revenue from './Revenue';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { compose } from 'redux';

const styles = () => ({
  red: {
    color: '#F44336',
  },
  green: {
    color: '#4CAF50',
  },
});

class Homepage extends Component {
  valueStatus = (value, success, alert) => (value > 0 ? success : alert);

  render() {
    const { budget, revenue, classes } = this.props;
    return (
      <Grid container spacing={0}>
        <Grid item xs={6}>
          <Card>
            <CardContent
              className={this.valueStatus(budget, classes.green, classes.red)}
            >
              <Typography type="display3" gutterBottom align="center">
                Budget
              </Typography>
              <Typography color="inherit" type="display2" align="center">
                <Budget />
              </Typography>
            </CardContent>
          </Card>
        </Grid>
        <Grid item xs={6}>
          <Card>
            <CardContent
              className={this.valueStatus(revenue, classes.green, classes.red)}
            >
              <Typography type="display3" gutterBottom align="center">
                Current revenue
              </Typography>
              <Typography color="inherit" type="display2" align="center">
                <Revenue />
              </Typography>
            </CardContent>
          </Card>
        </Grid>
      </Grid>
    );
  }
}

Homepage.propTypes = {
  budget: PropTypes.number.isRequired,
  revenue: PropTypes.number.isRequired,
  classes: PropTypes.object.isRequired,
};
const mapStateToProps = state => ({
  budget: state.budget,
  revenue: state.revenue,
});

export default compose(connect(mapStateToProps, null), withStyles(styles))(
  Homepage,
);
