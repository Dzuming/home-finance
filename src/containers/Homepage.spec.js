import React from 'react';
import { create } from 'react-test-renderer';
import Homepage from './Homepage';

const component = create(
   <Homepage/>
);

describe('<Homepage />', () => {
  it('should match snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

