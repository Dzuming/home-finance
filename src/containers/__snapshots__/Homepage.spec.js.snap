// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> should match snapshot 1`] = `
<div>
  <div
    className="MuiGrid-typeContainer-3"
  >
    <div
      className="MuiGrid-typeItem-4 MuiGrid-grid-xs-6-34"
    >
      <div
        className="MuiPaper-root-93 MuiPaper-shadow2-97 MuiPaper-rounded-94"
      >
        <div
          className="MuiCardContent-root-120 Homepage-green-2"
        >
          <h1
            className="MuiTypography-root-121 MuiTypography-display3-123 MuiTypography-gutterBottom-138 MuiTypography-alignCenter-134"
          >
            Budget
          </h1>
          <h1
            className="MuiTypography-root-121 MuiTypography-display2-124 MuiTypography-colorInherit-140 MuiTypography-alignCenter-134"
          >
            <div>
              123
               zł
            </div>
          </h1>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-typeItem-4 MuiGrid-grid-xs-6-34"
    >
      <div
        className="MuiPaper-root-93 MuiPaper-shadow2-97 MuiPaper-rounded-94"
      >
        <div
          className="MuiCardContent-root-120 Homepage-green-2"
        >
          <h1
            className="MuiTypography-root-121 MuiTypography-display3-123 MuiTypography-gutterBottom-138 MuiTypography-alignCenter-134"
          >
            Current revenue
          </h1>
          <h1
            className="MuiTypography-root-121 MuiTypography-display2-124 MuiTypography-colorInherit-140 MuiTypography-alignCenter-134"
          >
            <div>
              44
               zł
            </div>
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
